public/app/features/explore/TraceView/TraceView.tsx

```diff
@@ -68,6 +81,48 @@ export function TraceView(props: Props) {
    */
   const [slim, setSlim] = useState(false);
 
+  const panelState = useSelector((state: StoreState) => state.explore[props.exploreId]?.panelsState.trace);
+  const dispatch = useDispatch();
+  const focusedSpanId = panelState?.spanId;
+  const setFocusedSpanId = (spanId?: string) =>
+    dispatch(
+      changePanelState(props.exploreId, 'trace', {
+        ...panelState,
+        spanId,
+      })
+    );
+  const createFocusSpanLink = (traceId: string, spanId: string) => {
+    const link: DataLink<TempoQuery> = {
```

Comment:
@aocenas This code is hard-coding the assumption that your tracing datasource is tempo, which was safe for my use case but won't work in general.

The problem is that the the Trace Viewer needs to create an internal link to look up a particular traceId & spanId. That means you need to know what the trace datasource is & how to correctly structure a query to lookup by traceId. That might look different for zipkin, jaeger, x-ray datasources.

Reply:
Hmm I think we can sidestep this by just copying the current query. So something like:
```
internal: {
        datasourceUid: props.datasource.uid,
        datasourceName: props.datasource.name,
        query: explore[currentPane].query,
        panelsState: {
          trace: {
            spanId,
          },
        },
```
I think that should work well for all cases

Reply:
Smart, thanks!

Reply:
Fixed